src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -1075,6 +1088,371 @@ describe('Transactions Controller (Unit)', () => {
+        status: 500,
+      });
+
+      await request(app.getHttpServer())
+        .get(`/chains/${chainId}/safes/${safeAddress}/transactions/history`)
+        .expect(500)
+        .expect({
+          message: 'An error occurred',
+          code: 500,
+        });
+
+      expect(mockNetworkService.get).toBeCalledTimes(1);
+      expect(mockNetworkService.get).toBeCalledWith(
+        `${safeConfigApiUrl}/api/v1/chains/${chainId}`,
+        undefined,
+      );
+    });
+
+    it('Failure: Transaction API fails', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const chainResponse = chainBuilder().with('chainId', chainId).build();
+      mockNetworkService.get.mockResolvedValueOnce({ data: chainResponse });
+      mockNetworkService.get.mockRejectedValueOnce({
+        status: 500,
+      });
+
+      await request(app.getHttpServer())
+        .get(`/chains/${chainId}/safes/${safeAddress}/transactions/history/`)
+        .expect(500)
+        .expect({
+          message: 'An error occurred',
+          code: 500,
+        });
+
+      expect(mockNetworkService.get).toBeCalledTimes(2);
+      expect(mockNetworkService.get).toBeCalledWith(
+        `${safeConfigApiUrl}/api/v1/chains/${chainId}`,
+        undefined,
+      );
+      expect(mockNetworkService.get).toBeCalledWith(
+        `${chainResponse.transactionService}/api/v1/safes/${safeAddress}/all-transactions/`,
+        {
+          params: {
+            executed: true,
+            offset: PaginationData.DEFAULT_OFFSET,
+            limit: PaginationData.DEFAULT_LIMIT,
+            ordering: undefined,
+            queued: false,
+            safe: safeAddress,
+          },
+        },
+      );
+    });
+
+    it('Should return correctly each date label', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const chainResponse = chainBuilder().with('chainId', chainId).build();
+      const moduleTransaction = moduleTransactionTojson(
+        moduleTransactionBuilder()
+          .with('dataDecoded', null)
+          .with('executionDate', new Date('2022-12-06T17:09:36Z'))
+          .build(),
+      );
+      const multisigTransaction = multisigTransactionToJson(
+        multisigTransactionBuilder()
+          .with('dataDecoded', null)
+          .with('origin', null)
+          .with('executionDate', new Date('2022-12-25T17:09:36Z'))
+          .build(),
+      );
+      const transfer: any = nativeTokenTransferToJson(
+        nativeTokenTransferBuilder()
+          .with('executionDate', new Date('2022-12-31T17:09:36Z'))
+          .build(),
+      );
+      const incomingTransaction = ethereumTransactionToJson(
+        ethereumTransactionBuilder()
+          .with('executionDate', new Date('2022-12-31T17:09:36Z'))
+          .with('transfers', [transfer])
+          .build(),
```

Comment:
Is it possible to use faker/builder defaults to set the dates here? If we need to set the dates to a specific day we could use `faker.date.between` and reference the expected values in the assertions.

Reply:
Done 00777ace656c4400dd865c87065449d4c2a379b3